core/src/main/java/org/elasticsearch/index/query/CommonTermsQueryParser.java

```diff
@@ -87,7 +87,8 @@ public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOExcept
                         text = parser.objectText();
                     } else if ("analyzer".equals(currentFieldName)) {
                         analyzer = parser.text();
-                        if (parseContext.analysisService().analyzer(analyzer) == null) {
+                        //norelease we have to delay this check to the shard when parsing on the coordinating node
+                        if (parseContext.context().analysisService().analyzer(analyzer) == null) {
```

Comment:
maybe we should change this and move the check to the data node already? not sure //norelease is good here, or maybe I am missing something